buildSrc/src/main/java/UpdateContributorStatisticsTask.kt

```diff
@@ -0,0 +1,175 @@
+        ignoredCommits.forEach { println("https://github.com/streetcomplete/StreetComplete/commit/$it") }
+        println()
+        println("top 50 non-ignored commits with lines of code changed")
+        println("--------------------------------------------------------")
+        linesOfCodeChangedByCommit.entries.sortedByDescending { it.value }.subList(0, 50)
+            .forEach { println("${it.value}\thttps://github.com/streetcomplete/StreetComplete/commit/${it.key}") }
+        println()
+        println("top 25 non-ignored commits with most asset files changed")
+        println("--------------------------------------------------------")
+        assetFilesChangedByCommit.entries.sortedByDescending { it.value }.subList(0, 25)
+            .forEach { println("${it.value}\thttps://github.com/streetcomplete/StreetComplete/commit/${it.key}") }
+        println()
+        println("top 50 non-ignored commits with most lines of interface markup changed")
+        println("--------------------------------------------------------")
+        linesOfInterfaceMarkupChangedByCommit.entries.sortedByDescending { it.value }.subList(0, 50)
+            .forEach { println("${it.value}\thttps://github.com/streetcomplete/StreetComplete/commit/${it.key}") }
+
+        return result.values
+    }
+
+    private fun downloadGithubUserDetails(hash: String): GithubUser? {
+        val url = URL("https://api.github.com/repos/streetcomplete/streetcomplete/commits/$hash")
+        val conn = url.openConnection()
+        conn.setRequestProperty("Authorization", "token $githubApiToken")
+        val response = conn.getInputStream().bufferedReader().readText()
+        val jsonFormat = Json { ignoreUnknownKeys = true }
+        return jsonFormat.decodeFromString<GithubCommitDetails>(response).author
+    }
+
+    private fun mergeContributorsByGithubAccount(contributors: Collection<Contributor>): List<Contributor> {
+        val result = mutableListOf<Contributor>()
+        val contributorsByGithubUsername = contributors.groupBy { it.githubUsername }
+        for ((githubUsername, contributor) in contributorsByGithubUsername) {
+            result.add(Contributor(
+                name = contributor.maxByOrNull { it.lastChangeTimestamp }!!.name,
+                githubUsername = githubUsername,
+                linesOfCodeChanged = contributor.sumOf { it.linesOfCodeChanged },
+                linesOfInterfaceMarkupChanged = contributor.sumOf { it.linesOfInterfaceMarkupChanged },
+                assetFilesChanged = contributor.sumOf { it.assetFilesChanged }
+            ))
+        }
+        return result
+    }
+
+    private fun writeContributors(contributors: List<Contributor>) {
+        val jsonFormat = Json {
+            prettyPrint = true
+        }
+        val text = jsonFormat.encodeToString(ListSerializer(Contributor.serializer()), contributors)
+        File(targetFile).writeText(text)
+    }
+}
+
+@Serializable
+data class Contributor(
```

Comment:
Splitting this into two separate `GitContributor` and `GitHubContributor` with only the relevant (non-nullable) fields might make this easier to read. However, it's okay like this, too.

Reply:
Yeah, it's just a script after all. Not part of some bigger architecture where strict typing would lead to more clarity/less bugs